src/frontend-samples/screen-space-effects-sample/ScreenSpaceEffectsUI.tsx

```diff
@@ -0,0 +1,153 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { Angle } from "@bentley/geometry-core";
+import { IModelApp, ScreenViewport } from "@bentley/imodeljs-frontend";
+import "common/samples-common.scss";
+import { ControlPane } from "Components/ControlPane/ControlPane";
+import { ReloadableViewport } from "Components/Viewport/ReloadableViewport";
+import * as React from "react";
+import { Select, Toggle } from "@bentley/ui-core";
+import { effects, EffectsConfig, effectsConfig } from "./Effects";
+
+interface UIState {
+  // Index of the selected effect.
+  activeEffectIndex: number;
+  viewport?: ScreenViewport;
+  effectsConfig: EffectsConfig;
+  // Lens angle of the viewport's camera.
+  lensAngle: number;
+}
+
+interface UIProps {
+  iModelName: string;
+  iModelSelector: React.ReactNode;
+  effectNames: string[];
+}
+
+export default class ScreenSpaceEffectsUI extends React.Component<UIProps, UIState> {
+  public state: UIState = {
+    activeEffectIndex: 0,
+    effectsConfig,
+    lensAngle: 90,
+  };
+
+  private readonly _onChangeActiveEffect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState({ activeEffectIndex: Number.parseInt(event.target.value, 10) });
+  }
+
+  // Create a slider to adjust one of the properties of `EffectsConfig`.
+  private createSlider(label: string, value: number, min: number, max: number, step: number, update: (newValue: number) => void) {
+    const updateValue = (event: React.ChangeEvent<HTMLInputElement>) => {
+      update(Number(event.target.value));
+      this.setState({ effectsConfig });
+
+      // Ask the viewport to redraw its contents so that the changes to the effect are immediately visible.
+      if (this.state.viewport)
+        this.state.viewport.requestRedraw();
+    };
+
+    return (
+      <>
+        <span>{label}</span>
+        <input type="range" min={min} max={max} step={step} value={value} onChange={updateValue}></input>
```

Comment:
onChange only raises an event when the slider is finished moving.  Generally, onUpdate is better since it raises an event at every step, since the interaction is smoother.

Reply:
I'm not using a `Slider`, I'm using an `input`. I switched from `onChange` to `onInput`, which is supposed to update whenever the value changes, but it still seems to only invoke callback when slider stops moving.

Reply:
Just noticed you're using a basic html element and not ui-core Slider component.  Pushing fix.

Reply:
> > > @Ellord207 thumbnails acceptable?
> > 
> > 
> > They are both the correctly formatted and sized.
> > But I do have a complaint. Typically, the icon matches the view of the sample when it is first loaded.
> 
> That explains why most of the thumbnails look identical. I wanted to actually show the effect.

You can limit the list of selectable iModels, then when loading the view, select a customized camera position as you like it.

Reply:
Pushed fix, but it's still not as smooth as I would expect.  Is this just it's performance or would there be a way to demand a redraw quicker?

Reply:
`Viewport.requestRedraw` is supposed to request an animation frame. It doesn't. I pushed a workaround until that's fixed and fix integrated into the showcase.